swan-lake/learn-the-language/language-basics.md

```diff
@@ -22,7 +22,7 @@ This guide presents the subset of the Ballerina language that is common to most
 
 Ballerina programs consist of modules and each module consists of one or more *`.bal`* files.
   
-The most important thing that a module does is define named functions.
+The most important thing that a module does is, defining named functions.
```

Comment:
I think the original version is correct. Also did a quick google search for something like `"The most important thing that it does is"`, and IMO the hits indicate the original form is correct.

Reply:
Checked using Grammarly also.

Reply:
They're different tenses - defining is present continuous, which means the module is currently and frequently defining named functions and will continue to do so in the future but may soon finish. On the other hand, define is simple present tense, and is usually used for a fixed habit or routine so more suitable to indicate the function of the module.

The difference is really subtle though and really depends on the nature of the module. Both `define` and `defining` can be used in the sentence. The usage depends on the context. 

Define - Present
Defining - Present continuous

Grammarly suggestions might not consider the context correctly sometimes.

Rather, we can reword the sentence to avoid confusion as: `A module is mainly used to define named functions.`.

Reply:
With this kind (form) of sentence what I'm familiar with is the current version. 

Here we are not talking about the nature of a module and I don't think it is an issue with the tense. I don't see how it is context-specific either. 

This is just a way of specifying that modules define named functions.

Reply:
+1